framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4155,23 +4155,32 @@ VulkanReplayConsumerBase::OverrideCreateBuffer(PFN_vkCreateBuffer
 
     VkResult                              result = VK_SUCCESS;
     VulkanResourceAllocator::ResourceData allocator_data;
-    auto                                  replay_buffer      = pBuffer->GetHandlePointer();
-    auto                                  capture_id         = (*pBuffer->GetPointer());
-    auto                                  replay_create_info = pCreateInfo->GetPointer();
-
+    auto                                  replay_buffer               = pBuffer->GetHandlePointer();
+    auto                                  capture_id                  = (*pBuffer->GetPointer());
+    auto                                  replay_create_info          = pCreateInfo->GetPointer();
+    VkBufferCreateInfo                    modified_create_info        = *replay_create_info;
```

Comment:
We used to have a null check on line 4220 but now we trust this isn't null. Is that safe?